kalite/distributed/management/commands/setup.py

```diff
@@ -332,32 +332,29 @@ def handle(self, *args, **options):
 
         install_clean = not database_exists
 
-        if database_file:
-            if not database_exists:
-                install_clean = True
+        if database_exists:
+            # We found an existing database file.  By default,
+            #   we will upgrade it; users really need to work hard
+            #   to delete the file (but it's possible, which is nice).
+            logger.info(
+                "-------------------------------------------------------------------\n"
+                "WARNING: Database file already exists!\n"
+                "-------------------------------------------------------------------"
+            )
+            if not options["interactive"] \
+               or raw_input_yn("Keep database file and upgrade to KA Lite version %s? " % VERSION) \
+               or not raw_input_yn("Remove database file '%s' now? " % database_file) \
+               or not raw_input_yn("WARNING: all data will be lost!  Are you sure? "):
```

Comment:
It occurs to me that I need several minutes to understand what the above does and how `install_clean` is set.